### ISSUE TITLE
Version of Zxing?

### ISSUE THREAD

**Issue by seed93** [nu-book/zxing-cpp#82]
What is the commit of original zxing this project used?
I tried an image, which succeeded in https://zxing.org/w/decode.jspx, but failed here.
![图片](https://user-images.githubusercontent.com/4570032/61027377-917ae000-a3e8-11e9-8e84-4b35e64ebbd4.png)

**Comment by axxel** [comment]
It is basically from about 3 years ago but relevant changes have (mostly) been ported to this project since then.

I had a look at your specific example and could reproduce your results. I found that it works, if you crop your image to only contain the barcode-part:
![cropped-code128-works](https://user-images.githubusercontent.com/1038346/61170443-5d7af880-a569-11e9-9998-23de30c89f56.png)
The reason why it does not work without the cropping is because the `GlobalHistogramBinarizer` does not cope well with the additional 'non-code' pixels and produces a wrong black/white threshold. Interestingly, the relevant code is exactly the same (Java vs. C++).

TLDR; I don't know why it works with the upstream code but not with this fork.
